src/api-gateway.ts

```diff
@@ -1,23 +1,17 @@
-import {FunctionContext, FunctionEvent, FunctionResult} from './function';
+import type { Http } from "./http";
 
-export interface AuthorizerResult {
-    isAuthorized: boolean;
-    context: Record<string, any>
-}
-
-type PickedFunctionEventProps = Pick<FunctionEvent,
-    'httpMethod' |
-    'headers' |
-    'queryStringParameters' |
-    'requestContext'
->
+type PickedFunctionEventProps = Pick<
+    Http.Event,
+    "httpMethod" | "headers" | "queryStringParameters" | "requestContext"
+>;
 
-export interface AuthorizerEvent extends PickedFunctionEventProps {
-    resource: string;
-    cookies: Record<string, string>;
+export namespace ApiGateway {
+    export interface Event extends PickedFunctionEventProps {
```

Comment:
We should add to names of these two interfaces `Authorizer` word, since it is important to emphasise  that is not abstract ApiGateway related event, but it is call to function which is declared as authorizer for ApiGataeway.